x/retry/config.go

```diff
@@ -87,24 +88,25 @@ type MiddlewareConfig struct {
 
 // NewUnaryMiddlewareFromConfig creates a new policy provider that can be used
 // in retry middleware.
-func NewUnaryMiddlewareFromConfig(src interface{}, opts ...MiddlewareOption) (*OutboundMiddleware, error) {
+func NewUnaryMiddlewareFromConfig(src interface{}, opts ...MiddlewareOption) (*OutboundMiddleware, context.CancelFunc, error) {
```

Comment:
this is unfortunately a breaking API change, I couldn't think of a way around until pally is made available in the fx graph.  @akshayjshah thoughts?

Reply:
We are tightly linked to the two services that should be using this feature, so it should be nbd to update it.

Reply:
Could you add it to `OutboundMiddleware` somehow? I'm assuming you thought of this